ddl/partition.go

```diff
@@ -1134,22 +1134,28 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 		return ver, errors.Wrapf(err, "failed to notify PD the placement rules")
 	}
 
-	oldRules := make([]string, 0, len(oldIDs))
-	newRules := make([]*label.Rule, 0, len(oldIDs))
+	tableID := fmt.Sprintf(label.TableIDFormat, label.IDPrefix, job.SchemaName, tblInfo.Name.L)
+	oldPartRules := make([]string, 0, len(oldIDs))
 	for _, newPartition := range newPartitions {
-		oldRuleID := fmt.Sprintf(label.PartitionIDFormat, label.IDPrefix, job.SchemaName, tblInfo.Name.L, newPartition.Name.L)
-		oldRules = append(oldRules, oldRuleID)
+		oldPartRuleID := fmt.Sprintf(label.PartitionIDFormat, label.IDPrefix, job.SchemaName, tblInfo.Name.L, newPartition.Name.L)
+		oldPartRules = append(oldPartRules, oldPartRuleID)
 	}
 
-	rules, err := infosync.GetLabelRules(context.TODO(), oldRules)
+	rules, err := infosync.GetLabelRules(context.TODO(), append([]string{tableID}, oldPartRules...))
```

Comment:
I think `append(oldPartRules, tableID)` will  probably be more efficient than others.